### ISSUE TITLE
Investigate: currency model is not updated when `currentCurrency` is changed

### ISSUE THREAD

**Issue by PascalPrecht** [status-im/status-desktop#6092]
This is something I ran into when working on handling sync setting messages (#5201).

In `CurrencyStore` we have a `currentCurrency` property that represents the current default currency. When that property changes, we have a `*Changed` handler that ensures the currency model for the language & currency view is updated accordingly.

Here's what the handler looks like:

```qml
    onCurrentCurrencyChanged: { 
      updateCurrenciesModel()
    }
```

And `updateCurrenciesModel` looks like this:

```qml
    function updateCurrenciesModel() {
        var isSelected = false
        for(var i = 0; i < currenciesModel.count; i++) {
            if(root.currentCurrency === root.currenciesModel.get(i).key) {
              isSelected = true
                root.currenciesModel.get(i).selected = true
            }
            else {
                root.currenciesModel.get(i).selected = false
            }
        }

        // Set default:
        if(!isSelected)
            root.currenciesModel.get(0).selected = true
    }
```

When I was working on #5201 I got a signal from a service that the currency has changed (via sync), which then triggered a changes in the wallet section's `currentCurrency`, which then in turn is a change in `CurrencyStore.currentyCurrency`, which will ultimately trigger the `onCurrentCurrencyChanged` handler.

This is great, because we want the UI to update when there's such a change coming from the service.

Now the sad through is that we realized, even though the correct model is updated and set to `selected = true`, the currency selector in the settings view for languages and currency is **not** updated correctly.

This can be easily reproduced by adding the following snippet to the `LanguageView.qml`:

```qml
        StatusButton {
          text: "Test"
          onClicked: {
              root.currencyStore.currentCurrency = "eur"
          }
        }
```

^ Clicking this button will trigger the change in the currency, which will cause the currency model to update, however the update is not reflected in the view for some reason.

@noeliaSD has looked into this as well.

**Comment by borismelnik** [comment]
You need to try update model in `updateCurrenciesModel` like
```
...
    let obj = root.currenciesModel.get(i)
    obj.selected = true
    root.currenciesModel.set(i, obj)
...

```
it should call data changed event for all model users

**Comment by PascalPrecht** [comment]
@noeliaSD Mind if I assign you to this as you're most familiar with the listpicker component?

**Comment by noeliaSD** [comment]
> @noeliaSD Mind if I assign you to this as you're most familiar with the listpicker component?

Yeah! I forgot to do it!

**Comment by osmaczko** [comment]
> * Also the manual model filtering transfer is a maintenance burden. The usual C++ way is to build a proxy model, not sure if this can be easily done in NIM, but it is a more efficient way. Otherwise every change you do in the model you have to propagate it to delegates, as mentioned above

Exactly! Imperative model building/filtering is [asking for trouble](https://github.com/status-im/status-desktop/pull/5823#discussion_r885659791). I will take a look at the possibility to integrate `QSortFilterProxyModel` based utilities into StatusQ.

**Comment by osmaczko** [comment]
If I have to guess I would say this must be an issue similar to [this](https://github.com/status-im/status-desktop/pull/5823#discussion_r885659791). Manually built filtered model in `StatusListPicker` is not able to detect source model property changes :man_shrugging:

**Comment by anastasiyaig** [comment]
@alaibe please take a look, if this actual?
